SCC (SecurityContextConstraints) prevent clabernetes from working on OpenShift/OKD

```
Defaulted container "manager" out of: manager, init (init)
    INFO |               clabernetes | starting clabernetes...
    INFO |               clabernetes | initializing cluster info
    INFO |               clabernetes | begin prepare...
    INFO |               clabernetes | preparing certificates...
    INFO |               clabernetes | preparing scheme...
    INFO |               clabernetes | starting http manager...
    INFO |               clabernetes | start leader election
CRITICAL |              http-manager | http manager server has failed, error: open /clabernetes/certificates/webhook/tls.crt: permission denied
    INFO |                      klog | klog leaderelection.go:257: attempting to acquire leader lease c9s-system/clabernetes-manager...
E1026 13:32:49.695973       1 leaderelection.go:448] error retrieving resource lock c9s-system/clabernetes-manager: Get "https://172.30.0.1:443/apis/coordination.k8s.io/v1/namespaces/c9s-system/leases/clabernetes-manager": context canceled
    INFO |                      klog | klog leaderelection.go:448: error retrieving resource lock c9s-system/clabernetes-manager: Get "https://172.30.0.1:443/apis/coordination.k8s.io/v1/namespaces/c9s-system/leases/clabernetes-manager": context canceled
    INFO |                      klog | klog leaderelection.go:448: error retrieving resource lock c9s-system/clabernetes-manager: Get "https://172.30.0.1:443/apis/coordination.k8s.io/v1/namespaces/c9s-system/leases/clabernetes-manager": context canceled
    INFO |                      klog | klog leaderelection.go:448: error retrieving resource lock c9s-system/clabernetes-manager: Get "https://172.30.0.1:443/apis/coordination.k8s.io/v1/namespaces/c9s-system/leases/clabernetes-manager": context canceled
    INFO |               clabernetes | stopping clabernetes...
```

Error from the manager pod(s). This happens because (by default) the scc `restricted-v2` is applied which randomizes the uid of the default user inside the container.

This is not just limited to the manager pods, this is also an issue when a topology resource is created - the pods will *not* be created since by default, privileged pods are not allowed.

--- a/charts/clabernetes/templates/openshift.yaml
+++ b/charts/clabernetes/templates/openshift.yaml
@@ -1,0 +1,33 @@
+{{- if and (.Capabilities.APIVersions.Has "security.openshift.io/v1") .Values.manager.restrictedRBAC.enabled -}}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: anyuid-scc-rolebinding
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:openshift:scc:anyuid
+subjects:
+- kind: ServiceAccount
+  name: "{{ .Values.appName }}-service-account"
+  namespace: {{ .Release.Namespace }}
+---
+{{- range .Values.manager.restrictedRBAC.targetNamespaces -}}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: privileged-scc-rolebinding
+  namespace: {{ . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:openshift:scc:privileged
+subjects:
+- kind: ServiceAccount
+  name: clabernetes-launcher-service-account
+  namespace: {{ . }}
+---
+{{- end -}}
+{{- end -}}

--- a/charts/clabernetes/values.yaml
+++ b/charts/clabernetes/values.yaml
@@ -50,6 +50,7 @@ manager:
   # ClusterRole given to the manager's ServiceAccount is more limited, and a Role and
   # RoleBinding are created in each of the namespaces listed in `targetNamespaces` below to
   # grant extra permissions.
+  # This is needed on OpenShift clusters to prevent default SCC restrictions.
   restrictedRBAC:
     enabled: false
     targetNamespaces: []

